cmd/realm/triggers.go

```diff
@@ -69,111 +71,351 @@ var triggersDeleteCmd = &cobra.Command{
 		os.Exit(1)
 	}
 
-	utils.MaybeCurlAndExit(getTriggerCall, astarteAPIClient)
+	for _, name := range realmTriggers {
 
-	getTriggerRes, err := getTriggerCall.Run(astarteAPIClient)
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
+		triggerDefinition, err := getTriggerDefinition(realm, name)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		respJSON, err := json.MarshalIndent(triggerDefinition, "", "  ")
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		filename := fmt.Sprintf("/%s/%s.json", targetPath, name)
+		outFile, err := os.Create(filename)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+		defer outFile.Close()
+
+		if _, err := outFile.Write(respJSON); err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+	}
+	return nil
+}
+
+func triggersSyncF(command *cobra.Command, args []string) error {
+	if viper.GetBool("to-curl") {
+		fmt.Println(`'triggers sync' does not support the --to-curl option. Install your triggers one by one with 'triggers install'.`)
 		os.Exit(1)
 	}
 
-	triggerDefinition, _ := getTriggerRes.Parse()
-	respJSON, _ := json.MarshalIndent(triggerDefinition, "", "  ")
-	fmt.Println(string(respJSON))
+	triggersToInstall := []map[string]interface{}{}
+	triggersToUpdate := []map[string]interface{}{}
+	invalidTriggers := []string{}
+
+	for _, f := range args {
+		triggerFile, err := os.ReadFile(f)
+		if err != nil {
+			return err
+		}
+		if !validateTrigger(f) {
+			invalidTriggers = append(invalidTriggers, f)
+			continue
+		}
+
+		// var astarteTrigger map[string]triggers.Trigger{}
+		var astarteTrigger map[string]interface{}
+		if err = json.Unmarshal(triggerFile, &astarteTrigger); err != nil {
+			return err
+		}
+
+		if _, err := getTriggerDefinition(realm, astarteTrigger["name"].(string)); err != nil {
+			// The trigger does not exist
+			triggersToInstall = append(triggersToInstall, astarteTrigger)
+		} else {
+			triggersToUpdate = append(triggersToUpdate, astarteTrigger)
+		}
 
+		if len(triggersToInstall) == 0 && len(triggersToUpdate) == 0 {
+			// All good in the hood
+			fmt.Println("Your realm is in sync with the provided triggers files")
+			return nil
+		}
+
+	}
+	// Notify the user about what we're about to do
+	list := []string{}
+
+	for _, v := range triggersToInstall {
+		list = append(list, v["name"].(string))
+	}
+
+	list_existing := []string{}
+
+	for _, v := range triggersToUpdate {
+		list_existing = append(list_existing, v["name"].(string))
+	}
+
+	fmt.Printf("The following triggers are invalid and thus will not be processed: %+q \n", invalidTriggers)
+
+	//install new triggers
+	if len(triggersToInstall) > 0 {
+
+		fmt.Printf("The following new triggers will be installed: %+q \n", list)
+		fmt.Printf("\n")
+
+		if ok, err := utils.AskForConfirmation("Do you want to continue?"); !ok || err != nil {
+			fmt.Printf("aborting")
+			return nil
+		}
+
+		for _, v := range triggersToInstall {
```

Comment:
Since one of the fileds is used in the for body, give a meaningful name to `v` (e.g. `trigger`)